File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.java
import java.util.*;

class Solution {
    public int solution(int cacheSize, String[] cities) {
        Map<String, Integer> cache = new LinkedHashMap<>(cacheSize, 0.75f, true) {
            @Override
            protected boolean removeEldestEntry(Map.Entry eldest) {
                return size() > cacheSize;
             }
        };
        
        int answer = 0;
        for (String city : cities) {
            city = city.toLowerCase();
            if (cache.containsKey(city)) {
                answer += 1;
            } else {
                answer += 5;
            }
            cache.put(city, 1);
        }
        return answer;
    }
}

File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.py
from collections import deque

def solution(cacheSize, cities):
    cache, answer = deque(maxlen=cacheSize), 0
    for city in cities:
        city = city.lower()
        if city in cache:
            answer += 1
            cache.remove(city)
        else:
            answer += 5
        cache.append(city)
    return answer